qutip/qobjevo.py

```diff
@@ -139,12 +140,26 @@ def __call__(self, t, args={}):
 class _CubicSplineWrapper:
     # Using scipy's CubicSpline since Qutip's one
     # only accept linearly distributed tlist
-    def __init__(self, tlist, coeff):
+    def __init__(self, tlist, coeff, args=None):
         self.coeff = coeff
         self.tlist = tlist
-        self.func = CubicSpline(self.tlist, self.coeff)
+        try:
+            use_step_func = args["_step_func_coeff"]
+        except:
+            use_step_func = 0
+        if use_step_func:
+            self.func = interp1d(
+                self.tlist, self.coeff, kind="previous",
+                bounds_error=False, fill_value=0.)
+        else:
+            self.func = CubicSpline(self.tlist, self.coeff)
 
     def __call__(self, t, args={}):
+        try:
+            (self.func([t])[0])
+        except:
```

Comment:
If it sometime fails, better raise an exception.

Reply:
My mistake, used it for debugging, should be deleted